### PR TITLE
[EI_TOOL-64] Add "buildMessage" attribute to loadbalance and failover endpoints.

### PR DESCRIPTION
**Pull request by Bhashinee**
Enable adding "buildMessage" attribute to loadbalance and failover endpoints
from developer studio design view and source view.
Fixed: wso2/devstudio-tooling-ei#64

### CODE REVIEW

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src/org/wso2/developerstudio/eclipse/gmf/esb/provider/FailoverEndPointItemProvider.java [hunk 1]
```diff
@@ -61,11 +61,34 @@ public FailoverEndPointItemProvider(AdapterFactory adapterFactory) {
...
 	
 		return itemPropertyDescriptors;
 	}
 
 	/**
+	 * This adds a property descriptor for the Build Message feature.
+	 * <!-- begin-user-doc -->
+	 * <!-- end-user-doc -->
+	 * @generated NOT
+	 */
```
Comment (pamod) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src/org/wso2/developerstudio/eclipse/gmf/esb/provider/FailoverEndPointItemProvider.java]: Shall we elaborate the javaDocs by explaining the usage of parameters ?

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/FailoverEndpointDeserializer.java [hunk 2]
```diff
@@ -67,7 +68,9 @@ public FailoverEndPoint createNode(IGraphicalEditPart part,AbstractEndpoint obje
 		if(StringUtils.isNotBlank(failoverEndpoint.getName())){
              executeSetValueCommand(END_POINT__END_POINT_NAME, failoverEndpoint.getName());
 		}
-		
+
```
Comment (wggihan) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/FailoverEndpointDeserializer.java]: Can you please remove this empty line here and line 73?

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.persistence/src/org/wso2/developerstudio/eclipse/gmf/esb/internal/persistence/FailoverEndPointTransformer.java [hunk 3]
```diff
@@ -132,6 +132,8 @@ public FailoverEndpoint create(TransformationInfo info, FailoverEndPoint visualE
 			synapseFailEP.setName(name);
 		}
 
+		synapseFailEP.setBuildMessageAtt(visualEndPoint.isBuildMessage());
+
```
Comment (wggihan) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.persistence/src/org/wso2/developerstudio/eclipse/gmf/esb/internal/persistence/FailoverEndPointTransformer.java]: Remove those unnecessary empty lines from this PR.

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src/org/wso2/developerstudio/eclipse/gmf/esb/provider/FailoverEndPointItemProvider.java (hunk 1 above)
Comment (Bhashinee) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src/org/wso2/developerstudio/eclipse/gmf/esb/provider/FailoverEndPointItemProvider.java]: Fixed

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/FailoverEndpointDeserializer.java (hunk 2 above)
Comment (Bhashinee) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/FailoverEndpointDeserializer.java]: Fixed

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.persistence/src/org/wso2/developerstudio/eclipse/gmf/esb/internal/persistence/FailoverEndPointTransformer.java (hunk 3 above)
Comment (Bhashinee) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb.persistence/src/org/wso2/developerstudio/eclipse/gmf/esb/internal/persistence/FailoverEndPointTransformer.java]: Done

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src/org/wso2/developerstudio/eclipse/gmf/esb/provider/FailoverEndPointItemProvider.java (hunk 1 above)
Comment (Bhashinee) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src/org/wso2/developerstudio/eclipse/gmf/esb/provider/FailoverEndPointItemProvider.java]: Fixed

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src/org/wso2/developerstudio/eclipse/gmf/esb/provider/FailoverEndPointItemProvider.java [hunk 4]
```diff
@@ -61,11 +60,34 @@ public FailoverEndPointItemProvider(AdapterFactory adapterFactory) {
...
+	protected void addBuildMessagePropertyDescriptor(Object object) {
+		itemPropertyDescriptors.add
+			(createItemPropertyDescriptor
+				(((ComposeableAdapterFactory)adapterFactory).getRootAdapterFactory(),
+				 getResourceLocator(),
+				 getString("_UI_FailoverEndPoint_buildMessage_feature"),
+				 getString("_UI_PropertyDescriptor_description", "_UI_FailoverEndPoint_buildMessage_feature", "_UI_FailoverEndPoint_type"),
```
Comment (a5anka) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src/org/wso2/developerstudio/eclipse/gmf/esb/provider/FailoverEndPointItemProvider.java]: Can't we create a constant for string `"_UI_FailoverEndPoint_buildMessage_feature"` and reuse in line 79 and 80?

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/FailoverEndpointDeserializer.java [hunk 5]
```diff
@@ -67,7 +68,7 @@ public FailoverEndPoint createNode(IGraphicalEditPart part,AbstractEndpoint obje
 		if(StringUtils.isNotBlank(failoverEndpoint.getName())){
              executeSetValueCommand(END_POINT__END_POINT_NAME, failoverEndpoint.getName());
```
Comment (prabushi) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/FailoverEndpointDeserializer.java]: Shall we fix this formatting error?

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb/src/org/wso2/developerstudio/eclipse/gmf/esb/EsbPackage.java [hunk 6]
```diff
@@ -42000,6 +42029,17 @@
 	EReference getLoadBalanceEndPoint_MediatorFlow();
 
 	/**
+	 * Returns the meta object for the attribute '{@link org.wso2.developerstudio.eclipse.gmf.esb.LoadBalanceEndPoint#isBuildMessage <em>Build Message</em>}'.
```
Comment (prabushi) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb/src/org/wso2/developerstudio/eclipse/gmf/esb/EsbPackage.java]: These comments exceeds 120 indentation limit. Can we format this?

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src/org/wso2/developerstudio/eclipse/gmf/esb/provider/FailoverEndPointItemProvider.java (hunk 4 above)
Comment (Bhashinee) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src/org/wso2/developerstudio/eclipse/gmf/esb/provider/FailoverEndPointItemProvider.java]: These are generated codes. It is the default way defined in GMF.

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb/src/org/wso2/developerstudio/eclipse/gmf/esb/EsbPackage.java (hunk 6 above)
Comment (Bhashinee) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb/src/org/wso2/developerstudio/eclipse/gmf/esb/EsbPackage.java]: Generated codes from GMF. These code segments will be replaced in the same way when we generate it again even if we format them now.

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/FailoverEndpointDeserializer.java (hunk 5 above)
Comment (Bhashinee) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/FailoverEndpointDeserializer.java]: Fixed in this PR https://github.com/wso2/devstudio-tooling-esb/pull/272

**Pull request merged**